src/Modules/Build.lua

```diff
@@ -1257,6 +1293,12 @@ function buildMode:AddDisplayStatList(statList, actor)
 						})
 					end
 				end
+				if statData.warnFunc and statVal and ((statData.condFunc and statData.condFunc(statVal, actor.output)) or not statData.condFunc) then 
+					local b,v = statData.warnFunc(statVal)
+					if b then
+						InsertIfNew(self.controls.warnings.lines, v)
+					end
```

Comment:
All of the `warnFunc` variables can be simplified.  You can just have it return a string (or `nil`, if the condition is false), and Lua will let you have this code instead:

```suggestion
					local v = statData.warnFunc(statVal)
					if v then
						InsertIfNew(self.controls.warnings.lines, v)
					end
```

The functions would just be `warnFunc = function(v) return v < 0 and "Your unreserved Life is negative" end`